src/Microsoft.AspNetCore.Mvc.Razor/Internal/DefaultTagHelperComponentContextInitializer.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Reflection;
+using Microsoft.AspNetCore.Mvc.Razor.TagHelpers;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.AspNetCore.Mvc.ViewFeatures;
+using Microsoft.AspNetCore.Razor.TagHelpers;
+using Microsoft.Extensions.Internal;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Internal
+{
+    /// <summary>
+    /// Default implementation of <see cref="ITagHelperComponentContextInitializer"/>.
+    /// </summary>
+    public class DefaultTagHelperComponentContextInitializer : ITagHelperComponentContextInitializer
+    {
+        /// <inheritdoc />
+        public ITagHelperComponent InitializeViewContext(ITagHelperComponent tagHelperComponent, ViewContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            var propertiesToActivate = PropertyActivator<ViewContext>.GetPropertiesToActivate(
+                tagHelperComponent.GetType(),
+                typeof(ViewContextAttribute),
+                CreateActivateInfo);
```

Comment:
This needs to have some kind of caching. This will be called for each component for each request. A simple way to do this would be with a concurrent dictionary keyed on the type.

Reply:
Yeah @dougbu dropped by and we spoke about that offline. Something similar to the DefaultTagHelperFactory

Reply:
❄️  🐝 ans